blockchain/chain/src/store/chain_store.rs

```diff
@@ -423,6 +434,80 @@ where
     })
 }
 
+pub fn get_chain_message<DB>(db: &DB, key: &Cid) -> Result<Box<dyn Message>, Error>
```

Comment:
Yeah I thought since we already have the trait we could easily just call the methods on on the trait objects since they have getters. this is mapped form get_c_message in the lotus and from what I got the idea is that they are using a message type without caring about what type of concrete type it is a bit like a typeclass. I thought about an enum but it was a question of why use it when the trait already exists, didn't want to be reinventing the wheel but I can change that.

Reply:
I think it's fine the way it is now, can revisit that point specifically later. Was more thinking out loud for the second part as I setup the trait in anticipation of how it would be used, but has rarely been needed as such so far